_posts/tsukori/2016-04-29-finalprojectupdate4.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: post
+author: tsukori
+title: "Fourth Final Project Update" 
+--- 
+
+<iframe src="https://trinket.io/embed/python/cfd0381bba" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+So I currently have a lot of working pieces, they just need to get hooked together so they function correctly. The fun/frustrating part about this project is that a lot of stuff I thought I understood about python and python variables, I didn't so I'm having to think a lot more creatively and actually do what is "right" instead of trying to brute force things into working. My plan for getting all of the pieces and parts working together is to draw it out and write what needs to go where and what needs to be happening, plus I'm going to make a new trinket and reorganize my modules cause they are currently a giant mess. 
```

Comment:
Regarding variables, if you're going to need to access a variable inside a function, make sure to first create it **outside** the function.  Then `global` inside every function you need it in.

For example:

```python

random1 = 0
random2 = 0

def change_random1():
  global random1 # this brings random1 into this function's scope
  random1 = 1

def change_random2()
  global random1
  global random2
  random2 = random1 + 1
```

You may find it simpler to use one dictionary variable for this purpose.

```python
state = {}

def set_random1():
  global state
  state["random1"] = 1

def set_random2():
  global state
  state["random2"] = state["random1"] + 1
```

Good luck!